lucetc/src/decls.rs

```diff
@@ -383,7 +383,7 @@ impl<'a> ModuleDecls<'a> {
             }
             _ => Err(Error::Unsupported(
                 "lucetc only supports memory 0".to_string(),
-            ))?,
```

Comment:
I'm also surprised to learn that `Err(..)?` where a returned expression would be isn't an error! but I guess rustc figures `Err(..)?` won't continue, so for the purposes of inference the entire thing gets a type of `!` or something.